src/main/java/org/mockito/internal/junit/UniversalTestListener.java

```diff
@@ -34,6 +36,7 @@ public void testFinished(TestFinishedEvent event) {
         Collection<Object> createdMocks = mocks.keySet();
         //At this point, we don't need the mocks any more and we can mark all collected mocks for gc
         //TODO make it better, it's easy to forget to clean up mocks and we still create new instance of list that nobody will read, it's also duplicated
+        //TODO clean up all other state, null out stubbingLookupListener
```

Comment:
Interesting idea. StateMaster manages thread local state. In this case, it's the state of test listener that we want to keep clean. Hence, StateMaster does not really apply here.